### PR TITLE
Add mem.available chart to FreeBSD

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

Fixes https://github.com/netdata/netdata-cloud/issues/352

`mem.available` is one of those special charts that we use in the node list in the cloud. It is however missing from FreeBSD. This PR adds it, so that it can be displayed in the cloud.

The relevant alert is also changed to use the new chart.

##### Test Plan

Compile on FreeBSD. Note the new chart, and the alerts `ram_available` and `ram_in_use` should be displaying correct values.

### CODE REVIEW

**Review by thiagoftsm — APPROVED** [review]
I ran this PR on `FreeBSD 13.1-RELEASE-p0` and I also compiled this PR on `Slackware current` as parent. On both I could see the alarms and new charts, and the our charts are matching results given by other FreeBSD tool:

Screenshot_20220615_185759

**Comment by ilyam8** [comment]
Is the description valid for FreeBSD?

> ..., without causing swapping.

**Comment by ilyam8** [comment]
@MrZammler I think we need to modify [mem.available](https://github.com/netdata/netdata/blob/d73e6c9890777732b7cfd106f1375e33c6c48377/web/gui/dashboard_info.js#L1736) to:

```js
    'mem.available': {
        info: function (os) {
            if (os === 'linux') {
                return 'Available Memory is estimated by the kernel, as the amount of RAM that can be used by userspace processes, without causing swapping.'
            }
    },
```

**Comment by thiagoftsm** [comment]
> Is the description valid for FreeBSD?
> 
> > ..., without causing swapping.
> 
> edit: oh, I think it is not - it described Linux MemAvailable.

FreeBSD also has [swap memory](https://people.freebsd.org/~blackend/en_US.ISO8859-1/books/handbook/adding-swap-space.html) and  remember it works more or less like Linux.

**Comment by ilyam8** [comment]
@thiagoftsm my point is Linux MemAvailable is a magic number calculated by the kernel, we can't apply the same description for the new metric.

**Comment by MrZammler** [comment]
> @thiagoftsm my point is Linux MemAvailable is a magic number calculated by the kernel, we can't apply the same description for the new metric.

Ok @ilyam8 thanks, any suggestions of what it might say? Something e.g. like:

`The sum of free, cached and inactive memory, that generally is available to userspace processes without causing swapping.`

**Comment by ilyam8** [comment]
@MrZammler something like this, yes

> The amount of memory available to user-space processes without swapping. Calculated as the sum of free, cached, and inactive memory.

**Comment by MrZammler** [comment]
Ok, changed the info field, but since the old message was the default for every case, I thought it's better to keep it still as default and only differentiate when on freebsd.

File: collectors/freebsd.plugin/freebsd_sysctl.c [hunk 1]
```diff
@@ -1055,6 +1055,34 @@ int do_system_ram(int update_every, usec_t dt) {
...
+                    "Available RAM for applications",
+                    "MiB",
+                    "freebsd.plugin",
+                    "system.ram",
+                    NETDATA_CHART_PRIO_MEM_SYSTEM_AVAILABLE,
+                    update_every,
+                    RRDSET_TYPE_AREA
+            );
+
+            rd_avail   = rrddim_add(st_mem_available, "MemAvailable", "avail", system_pagesize, MEGA_FACTOR, RRD_ALGORITHM_ABSOLUTE);
```
[the remark this answers is not in the document]
Comment (ilyam8) [reply collectors/freebsd.plugin/freebsd_sysctl.c:1075]: Ok, we do the same for Linux. Should be fixed everywhere later.

**Review by ilyam8 — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
I am approving this PR without retest, because last commit addressed only dashboard. LGTM!

**Pull request merged**